### PR TITLE
don't set flag to "converged" if max iter exceeded

### LINKED ISSUE #9254: BUG: root finders brentq, etc, flag says "converged" even if maxiter exceeded
In version 1.1.0, and in master as of f3e0dcf70387c8ca4aecd83f57260a4f5bde82d1 there is no difference in the `RootResults.flag` output for a solution that exceeds the maximum iterations, `maxiter` and a successful solution, when setting arguments `full_output=True, disp=False`. If the max iterations were exceeded and setting args `full_output=True, disp=True` the same solution would raise a `RuntimeError`

In master as of f3e0dcf70387c8ca4aecd83f57260a4f5bde82d1 if `newton` were called with args `full_output=True, disp=False`, then `RootResults.flag` would say ["convergence error"](../blob/f3e0dcf70387c8ca4aecd83f57260a4f5bde82d1/scipy/optimize/zeros.py#L314) or raise an exception if `disp=True`.

In version 1.1.0, the error values of the [`flag_map`](https://github.com/scipy/scipy/blob/v1.1.0/scipy/optimize/zeros.py#L17) are never used, only "converged" is used.

In master as of f3e0dcf70387c8ca4aecd83f57260a4f5bde82d1 only `newton` uses the other values of the `flag_map` and the `SIGNERR` condition is never used.

In [`zeros.c` the flag is set to zero](../blob/master/scipy/optimize/zeros.c#L68) which is the only condition ever returned, because [in the case of a sign error, an exception is raised](../blob/f3e0dcf70387c8ca4aecd83f57260a4f5bde82d1/scipy/optimize/zeros.c#L118-L122) and in the case of a convergence error, if `disp==True` then an exception is raised. Inside the block `if params,error_num == CONVERR`, [the flag is temporarily set to 1](../blob/master/scipy/optimize/zeros.c#L130), but then never used. Perhaps this was unintentional? Or perhaps I misunderstand the purpose of this code. Perhaps the flag was meant to be set to `CONVERR`, ie: -2, and returned _outside_ of the if-block?

### Reproducing code example:
```python
from scipy.optimize import brentq
brentq(lambda x: 1.2*x**3 - 2.3*x**2 + 3.4*x -4.5, -30, 30, (), 1e-6, 1e-6, 10,
    full_output=True, disp=False)

(1.6062150063585068,       converged: True
            flag: 'converged'
  function_calls: 12
      iterations: 10
            root: 1.6062150063585068)
```
The only difference is the arg `disp=True`, now it doesn't converge.
```python
brentq(lambda x: 1.2*x**3 - 2.3*x**2 + 3.4*x -4.5, -30, 30, (), 1e-6, 1e-6, 10,
    full_output=True, disp=True)

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-36-0944cf7888ac> in <module>()
----> 1 brentq(lambda x: 1.2*x**3 - 2.3*x**2 + 3.4*x -4.5, -30, 30, (), 1e-6, 1e-6, 10, full_output=True, disp=True)

/usr/local/Cellar/python/3.6.4_4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/scipy/optimize/zeros.py in brentq(f, a, b, args, xtol, rtol, maxiter, full_output, disp)
    508     if rtol < _rtol:
    509         raise ValueError("rtol too small (%g < %g)" % (rtol, _rtol))
--> 510     r = _zeros._brentq(f,a,b,xtol,rtol,maxiter,args,full_output,disp)
    511     return results_c(full_output, r)
    512 

RuntimeError: Failed to converge after 10 iterations.
```
I think the issue is the Python extension returns zero, no matter what.
```python
from scipy.optimize import _zeros

_zeros._brentq(lambda x: 1.2*x**3 - 2.3*x**2 + 3.4*x -4.5, -30, 30, 1e-6, 1e-6, 10, (), True, False)

(1.6062150063585068, 12, 10, 0)
```
The only difference is the arg `disp=True`, now it doesn't converge.
```python
_zeros._brentq(lambda x: 1.2*x**3 - 2.3*x**2 + 3.4*x -4.5, -30, 30, 1e-6, 1e-6, 10, (), True, True)

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-38-5d64209d315a> in <module>()
----> 1 _brentq(lambda x: 1.2*x**3 - 2.3*x**2 + 3.4*x -4.5, -30, 30, 1e-6, 1e-6, 10, (), True, True)

RuntimeError: Failed to converge after 10 iterations.
```

### Error message:
I would expect `brentq`, etc. to return a flag that says "convergence error" with args `full_output=True, disp=False`.

### Scipy/Numpy/Python version information:
```python
import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.1.0 1.15.0 sys.version_info(major=3, minor=6, micro=4, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by mikofski**
* be consistent with underlying functions `brentq`, `bisection`, `ridder` that return `error_num` of `CONVERR` when maximum iteration exceeded, not "converged"
* be consistent with Newton and TOMS748 also return `_CONVERR` if maximum iterations exceeded instead of "converged"
* closes #9254
* use macro `CONVERGED` to be explicit in test for non-convergence
* remove unused line `flag = 1;` before `return NULL` which raises an exception, so flag never used
* set `flag = CONVERR` to be consistent with underlying functions in `Zeros/` and Newton and TOMS
* explicitly set `flag = CONVERGED` if test for non-convergence fails, meaning successful convergence

### CODE REVIEW

**Comment by mikofski** [comment]
@pvanmulbregt what do you think about making Brent, bisection, and Ridder self-consistent and consistent with Newton and TOMS748? IE: if they quit because they reach the maximum iterations, instead of setting their full output flag to "converged" they should return `CONVERR`. That way a user won't think that they have a solution, instead of realizing that it actually just hit the iteration limit, and that they should restart the solver or reset the limit to a larger number.

File: scipy/optimize/zeros.c [hunk 1]
```diff
@@ -129,11 +129,12 @@ call_solver(solver_type solver, PyObject *self, PyObject *args)
                         "Failed to converge after %d iterations.",
                         solver_stats.iterations);
                 PyErr_SetString(PyExc_RuntimeError, msg);
-                flag = 1;
                 return NULL;
             }
+            flag = CONVERR;
```
Comment (pvanmulbregt) [inline scipy/optimize/zeros.c]: Approve of the use of the #define values `CONVERGED`, `CONVERR` etc.
It looks as though the value of `flag` is set on line 134 to `CONVERR` but is overridden just below on line 137 before this value has a chance to be used?

Comment (mikofski) [reply scipy/optimize/zeros.c]: * moved line 137 into `else{}`, _ie_: if not converged then must be converged
* also added test to check this works

File: scipy/optimize/tests/test_zeros.py [hunk 2]
```diff
@@ -661,3 +661,24 @@ def fpp_array(x):
...
+        lambda x: ((1.2*x - 2.3)*x + 3.4)*x - 4.5,
+        -30, 30, (), 1e-6, 1e-6, maximum_iterations,
+        full_output=True, disp=False)
+    assert r[1].flag == zeros.CONVERR
+    assert r[1].iterations == maximum_iterations
+    more_maximum_iterations = 100
+    r = zeros.brentq(
+        lambda x: ((1.2*x - 2.3)*x + 3.4)*x - 4.5,
+        -30, 30, (), 1e-6, 1e-6, more_maximum_iterations,
+        full_output=True, disp=False)
```
Comment (tylerjereddy) [inline scipy/optimize/tests/test_zeros.py]: These `brentq` calls are duplicates except for the iteration counts. I haven't been "enforcing" parametrization with pytest too much though, so minor point.

**Review by tylerjereddy — COMMENTED** [review]
Makes sense to me from a high level. I suppose this isn't considered a compatibility break because returning a false indicator of convergence is effectively a bug.

@pvanmulbregt Are you +1 here now?

**Comment by mikofski** [comment]
I don't think [this failure](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=1129&view=results) is related, maybe it timed out?
[image](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=1129&view=results)

**Comment by tylerjereddy** [comment]
Hmm, hopefully a fluke. Probably better to rebase than merge master into this PR though--I don't think we want to have synchronization commits in our history if possible.

File: scipy/optimize/zeros.c [hunk 3]
```diff
@@ -129,11 +129,15 @@ call_solver(solver_type solver, PyObject *self, PyObject *args)
...
         }
     }
+    else
```
Comment (pvanmulbregt) [inline scipy/optimize/zeros.c]: `else` and `LEFT_BRACE` on the same line would match the style used in the rest of the file.

```
else {
    flag = CONVERGED;
}
```

File: scipy/optimize/tests/test_zeros.py (hunk 2 above)
Comment (pvanmulbregt) [reply scipy/optimize/tests/test_zeros.py]: I'm +1 (modulo the style nit pick in the block setting the flag to CONVERGED.)

**Comment by tylerjereddy** [comment]
Thanks @mikofski and @pvanmulbregt 

@mikofski can you add a release note here please: https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.3.0

**Pull request merged**